src/routes/hydra.ts

```diff
@@ -0,0 +1,253 @@
+import {NextFunction, Request, Response} from 'express'
+import config, {logger} from '../config'
+import {PublicApi, AdminApi as KratosAdminApi, Session } from '@oryd/kratos-client'
+import {AdminApi as HydraAdminApi, AcceptConsentRequest, AcceptLoginRequest, RejectRequest} from '@oryd/hydra-client'
+import url from 'url';
+
+const hydraAdminEndpoint = new HydraAdminApi(process.env.HYDRA_ADMIN_URL)
+const kratosPublicEndpoint = new PublicApi(config.kratos.public)
+const kratosAdminEndpoint = new KratosAdminApi(config.kratos.admin)
+
+type UserRequest = Request & { user: any }
+
+export default (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  // The request is used to identify the login and registration request in ORY Kratos and return data like the csrf_token and so on.
+  const request = req.query.request
+  const currentLocation = `${req.protocol}://${req.headers.host}${req.url}`;
```

Comment:
Again as in #60 this thing is not reverse-proxy - friendly, I've tried it with Jeremy's fork.
The situation is:
1. Reverse-proxy receives https response and redirects to selfservice using http.
2. Selfservice builds currentLocation with req.proto=http and after performing some more stuff sends resulting URL to Kratos as a final redirect location.
3. Kratos tells "No, http://selfservice is not whitelisted as opposed to https://selfservice"

Actually the solution would be to implement an env var from which selfservice will know it's own predefined base URL including the protocol.

But as for me supporting HTTPS also wouldn't be an odd option - Probably I'll try to implement it.

Reply:
Ok so if we implement #60 here the problem should go away?

Reply:
Not actually. I've just meant that this problem probably also will not occur on port-based configurations, just like the problem #60.

The problem will become overridable by config if we implement #62.

Reply:
And... It will definitely still exist if we still proxy https->http. I mean when we don't use config proposed in #62.
The solution could be implemented with an additional env var, for example:
`BASE_URL = https://selfservice.domain`
or
`REDIRECT_PROTO = https`

Actually there are lots of variations of fixing it. And I think it's not a blocker for this MR actually. As for me - #62 would be enough. Because if You actually can use HTTPS - why would you want to proxy HTTPS->HTTP?

Reply:
Oh, there is already a `BASE_PATH` env var. I should have read the readme file alittle bit earlier than while editing it :-)
It could definitely be embraced at this line.

Reply:
I mean smth like:
```
if (config.baseUrl) {
 currentLocation = `${config.baseUrl} .....`
} else {
  currentLocation = `${req.protocol} .....`
}
```

Reply:
Should be fixed with 2331d92